sds_data_manager/stacks/sds_data_manager_stack.py

```diff
@@ -136,6 +154,26 @@ def __init__(
             resources=["*"],
         )
 
+        snapshot_role_policy = iam.PolicyStatement(
+            effect=iam.Effect.ALLOW,
+            actions=[
+                "s3:ListBucket",
+                "s3:GetObject",
+                "s3:PutObject",
+                "s3:DeleteObject",
```

Comment:
should we have ability to delete on this bucket? Is it safe?

Reply:
From what I read, it seems like this is needed for snapshots to work. I can try removing it and see what happens, but in the documentation it says to setup the policy with delete permissions: https://docs.aws.amazon.com/opensearch-service/latest/developerguide/managedomains-snapshots.html

Reply:
interesting. Thank you for clarifying